pkg/kf/doctor/cluster.go

```diff
@@ -20,37 +20,31 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-// KubernetesClientFactory creates a Kubernetes client
-type KubernetesClientFactory func() (kubernetes.Interface, error)
-
 // ClusterDiagnostic tests that the cluster's Kubernetes version and components
 // are suitable for kf.
 type ClusterDiagnostic struct {
-	createKubernetesClient KubernetesClientFactory
+	c kubernetes.Interface
```

Comment:
I'd like to keep these named something that helps you understand what's going on if you're reading the rest of the code because `c.c.Discovery()` is a little opaque. Maybe `client` or `kubeclient`?